### PR TITLE
Lml/jit decorator

### PR DESCRIPTION
**Pull request by lml131**
1、rename parrots.jit.pat as mmcv.jit
2、add tests for mmcv.jit, these test cases will pass under both pytorch and parrots environment

### CODE REVIEW

File: mmcv/utils/parrots_jit.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            return wrapper
+        else:
+            return func
+
+
+if TORCH_VERSION == 'parrots':
+    from parrots.utils.tester import skip_no_elena
+else:
+
+    def bypass_decorator(func):
```
Comment (ZwwWayne) [inline mmcv/utils/parrots_jit.py:35]: Can we simply use `def skip_no_elena(func)` here?

**Comment by ZwwWayne** [comment]
LGTM now, please @hellock have a check.

**Comment by hellock** [comment]
Task linked: [CU-5brfzn MMCV add parrots JIT API](https://app.clickup.com/t/5brfzn)

File: mmcv/utils/parrots_jit.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import pytest
+import torch
+
+TORCH_VERSION = torch.__version__
+
+if TORCH_VERSION == 'parrots':
+    from parrots.jit import pat
+    jit = pat
```
Comment (hellock) [inline mmcv/utils/parrots_jit.py:8]: `from parrots.jit import pat as jit`

File: mmcv/utils/parrots_jit.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+        def wrapper(*args, **kargs):
+            return func(*args, **kargs)
+
+        return wrapper
+
+    skip_no_elena = bypass_decorator
+
+
+def use_parrots():
```
Comment (hellock) [inline mmcv/utils/parrots_jit.py:45]: `is_using_parrots`

File: mmcv/utils/parrots_jit.py (hunk 1 above)
Comment (lml131) [reply mmcv/utils/parrots_jit.py:35]: At first, this flag comes from the cmake compilation option when compiling parrots, and then it is converted into a macro definition in C + +. Finally, it is transformed into an interface visible in Python through pybind11. I don't find a method to define this interface simply in mmcv.

File: mmcv/utils/parrots_jit.py (hunk 2 above)
Comment (lml131) [reply mmcv/utils/parrots_jit.py]: I have updated code.

File: mmcv/utils/parrots_jit.py (hunk 3 above)
Comment (lml131) [reply mmcv/utils/parrots_jit.py]: I have updated code.

File: mmcv/utils/parrots_jit.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+else:
+
+    def bypass_decorator(func):
+
+        def wrapper(*args, **kargs):
+            return func(*args, **kargs)
+
+        return wrapper
+
+    skip_no_elena = bypass_decorator
```
Comment (hellock) [inline mmcv/utils/parrots_jit.py:41]:
```python
if TORCH_VERSION == 'parrots':
    from parrots.utils.tester import skip_no_elena
else:

    def skip_no_elena(func):

        def wrapper(*args, **kargs):
            return func(*args, **kargs)

        return wrapper
```

Comment (lml131) [reply mmcv/utils/parrots_jit.py:41]: ok

**Review by ZwwWayne — APPROVED** [review]

**Comment by OpenMMLab-Assistant-007** [comment]
Hi!
@lml131 
First of all, we want to express our gratitude for your significant PR in the OpenMMLab project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**